functions/zoom-downloader.py

```diff
@@ -140,22 +149,73 @@ def uuid(self):
     def zoom_series_id(self):
         return self.data["zoom_series_id"]
 
+    @property
+    def duration(self):
+        return int(self.data.get("duration", 0))
+
     @property
     def recording_files(self):
         if not hasattr(self, "_recording_files"):
 
+            # all incoming files
             files = self.data["recording_files"]
-            start_times = sorted(set(
+
+            # number of distinct start times is the count of track sets
+            track_set_start_times = sorted(set(
                 [file["recording_start"] for file in files]
             ))
 
+            # collect the recording files here as ZoomFile objs
             zoom_files = []
-            for file in files:
-                track_seq = start_times.index(file["recording_start"])
-                file["meeting_uuid"] = self.uuid
-                file["zoom_series_id"] = self.zoom_series_id
-                file["created_local"] = self._created_local
-                zoom_files.append(ZoomFile(file, track_seq))
+
+            # we'll set this to true once we're past any false start
+            # track sets
+            take_all = False
+
+            # track sets refers to the sets of recording files that
+            # are generated when a host stops and restarts a meeting
+            # Here we group the files by track set to allow ignoring any
+            # too-short segments that happen at the beginning of the meeting
+            # a.k.a. false starts
+            for track_set_num, start_time in enumerate(track_set_start_times):
+                # all the files from this segment
+                track_set_files = [f for f in files
+                             if f["recording_start"] == start_time]
+                # parse their start/end times
+                track_set_start = datetime.strptime(
+                    track_set_files[0]["recording_start"],
+                    TIMESTAMP_FORMAT
+                )
+                track_set_end = datetime.strptime(
+                    track_set_files[0]["recording_end"],
+                    TIMESTAMP_FORMAT
+                )
+                # get the duration of the segment
+                duration_sec = (track_set_end - track_set_start).seconds
+
+                logger.info({
+                    "track_set": {
+                        "number": track_set_num,
+                        "file_count": len(track_set_files),
+                        "start": track_set_start,
+                        "end": track_set_end,
+                        "duration": duration_sec
+                    }
+                })
+
+                # skip any initial track sets < minimum duration
+                if not take_all and duration_sec < MINIMUM_DURATION * 60:
+                    logger.info("skipping track set {}".format(track_set_num))
```

Comment:
I get what you're saying. We can revisit this if turns out to be a problem. I'm not going to do it now because in the thing i sent around and producers signed off on i clearly stated we'd only remove them from the beginning and I'm not going to a) do another round of sign-off, or b) release something that doesn't do what we asked them to confirm (even if it's a pretty insignificant difference).